core/common/src/test/java/alluxio/ConfigurationDocGeneratorTest.java

```diff
@@ -0,0 +1,157 @@
+/*
+ * The Alluxio Open Foundation licenses this work under the Apache License, version 2.0
+ * (the "License"). You may not use this work except in compliance with the License, which is
+ * available at www.apache.org/licenses/LICENSE-2.0
+ *
+ * This software is distributed on an "AS IS" basis, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND,
+ * either express or implied, as more fully set forth in the License.
+ *
+ * See the NOTICE file distributed with this work for information regarding copyright ownership.
+ */
+
+package alluxio;
+
+import static org.junit.Assert.assertEquals;
+
+import org.junit.Assert;
+import org.junit.Before;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.rules.TemporaryFolder;
+
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.nio.charset.StandardCharsets;
+import java.util.Collection;
+import java.util.List;
+import java.util.ArrayList;
+
+/**
+ * Tests for {@link ConfigurationDocGenerator}.
+ */
+public class ConfigurationDocGeneratorTest {
+  private static final String FILE_HEADER = "propertyName,defaultValue";
+  private String mLocation;
+
+  /** Rule to create a new temporary folder during each test. */
+  @Rule
+  public TemporaryFolder mFolder = new TemporaryFolder();
+
+  /**
+   * Sets up all dependencies before a test runs.
+   */
+  @Before
+  public void backupCSVFiles() throws Exception {
+    mLocation = mFolder.newFolder().toString() + "/";
+  }
+
+  public void checkFileContents(String source, List<String> target) throws Exception {
+    //assert file contents
+    assertEquals(2, target.size());
+    assertEquals(FILE_HEADER, target.get(0));
+    assertEquals(source, target.get(1));
+  }
+
+  @Test
+  public void checkCSVFile_user() throws Exception {
+    String key = "alluxio.user.local.reader.packet.size.bytes";
+    PropertyKey userLocalWriterPacketSizeBytes = new PropertyKey(key);
+    String defaultValue = Configuration.get(userLocalWriterPacketSizeBytes);
+    Collection<PropertyKey> defaultKeys = new ArrayList<>();
+    defaultKeys.add(userLocalWriterPacketSizeBytes);
+    ConfigurationDocGenerator.writeCSVFile(defaultKeys, mLocation);
+    String filePath = mLocation + "user-configuration.csv";
```

Comment:
please use `PathUtils.concatPath(mLocation, "user-configuration.csv")` 
Reason is that in this way, our code is more portable.